services/src/main/java/org/keycloak/services/clientregistration/oidc/DescriptionConverter.java

```diff
@@ -260,12 +264,19 @@ private static boolean setPublicKey(OIDCClientRepresentation clientOIDC, ClientR
                 CertificateInfoHelper.updateClientRepresentationCertificateInfo(clientRep, rep, JWTClientAuthenticator.ATTR_PREFIX);
 
                 configWrapper.setUseJwksUrl(false);
+                configWrapper.setUseJwksString(false);
 
                 return true;
             }
-        } else {
+        } else if (clientOIDC.getJwksUri() != null) {
             configWrapper.setUseJwksUrl(true);
             configWrapper.setJwksUrl(clientOIDC.getJwksUri());
+            configWrapper.setUseJwksString(false);
+            return true;
+        } else {
+            configWrapper.setUseJwksString(true);
```

Comment:
My vote is to change whole if/else block to something like this (Not 100% sure it is syntactically correct, but I hope it will help to illustrate what I mean). In other words remove whole first block of this if/else block and in case that OIDC representation has "jwks" set, then just set "useJwksString" to true in Keycloak representation. I think this will allow to remove `useJwksString` field from OIDCClientRepresentation as pointed above.
```
        if (clientOIDC.getJwks() != null) {
           configWrapper.setUseJwksString(true);
            configWrapper.setJwksString(clientOIDC.getJwksString());
            configWrapper.setUseJwksUrl(false);
            return true;
            }
        } else {
            configWrapper.setUseJwksUrl(true);
            configWrapper.setJwksUrl(clientOIDC.getJwksUri());
            return true;
        }
```

Reply:
@mposolda I changed the whole if/else block. Is my understanding correct?